### PR TITLE
FileStore: use pwritev instead of lseek+writev

### PR DESCRIPTION
**Pull request by yuyuyu101**

### CODE REVIEW

**Comment by yuyuyu101** [comment]
update on #6718 @jack-changtao plz review

**Comment by tchaikov** [comment]
@yuyuyu101 could you help @jack-changtao to squash his two commits into a single one?

File: src/common/buffer.cc [hunk 1]
```diff
@@ -1952,57 +1952,107 @@ int buffer::list::write_file(const char *fn, int mode)
...
-	  start[0].iov_base = (char *)start[0].iov_base + wrote;
-	  bytes -= wrote;
-	}
-	goto retry;
-      }
-      iovlen = 0;
-      bytes = 0;
+
+
+    int r = do_writev(fd, iov, offest, iovlen, bytes);
```
Comment (tchaikov) [inline src/common/buffer.cc]: s/offeset/offset/

File: src/include/config-h.in.cmake [hunk 2]
```diff
@@ -123,6 +123,9 @@
 /* Define to 1 if you have the `syncfs' function. */
 #cmakedefine HAVE_SYNCFS 1
 
+/* Define to 1 if you have the `pwritev' function. */
+#cmakedefine HAVE_PWRITEV 1
```
Comment (tchaikov) [inline src/include/config-h.in.cmake]: do you want to add
```
CHECK_FUNCTION_EXISTS(pwritev HAVE_PWRITEV) 
```
in `CMakeLists.txt` also?

**Comment by tchaikov** [comment]
@yuyuyu101 could you reconsider https://github.com/ceph/ceph/pull/7349#discussion_r50792617 ? w.r.t `CHECK_FUNCTION_EXISTS(pwritev HAVE_PWRITEV) ` otherwise your be6dbcd might not work as expected.

**Comment by yuyuyu101** [comment]
@tchaikov thanks! I missed it

**Comment by liewegas** [comment]
The write_fd changes break the crush map check on pool creation ('ceph osd pool create foo 12' fails).  Probably because we are doing write_fd to a pipe to the child process?

**Comment by yuyuyu101** [comment]
@liewegas updated, revert orignal write_fd

**Comment by liewegas** [comment]
@yuyuyu101 do you mind squashing this down so that write_fd isn't changed and then changed back?

**Comment by yuyuyu101** [comment]
@liewegas done

**Comment by liewegas** [comment]
unittest_bufferlist (and some other make check tests) fail

**Comment by yuyuyu101** [comment]
@liewegas  eh, pass now...

**Comment by yuyuyu101** [comment]
retest this please

**Pull request merged**